### PR TITLE
Set Luacheck globals from editor diagnostics

### PR DESCRIPTION
**Pull request by britzl**

### CODE REVIEW

File: script/core/diagnostics/luacheck.lua [hunk 1]
```diff
@@ -39,6 +29,59 @@ files.isLua = function(uri)
     return isLua(uri)
 end
 
+local defold_config = {
```
Comment (britzl) [inline script/core/diagnostics/luacheck.lua]: Not really sure if we need this tbh? We get the globals from the editor. Not really any point in also including them here.

Comment (britzl) [reply script/core/diagnostics/luacheck.lua]: I did note that what the editor provides also contains messages such as "set_camera" and it is missing the new "fixed_update" lifecycle function

Comment (vlaaad) [reply script/core/diagnostics/luacheck.lua]: Yeah, we probably don't need the `defold_config`! Yes, I also should exclude messages from globals. Maybe `fixed_update` is missing in sdoc where I get the globals from?

**Review by vlaaad — COMMENTED** [review]
Found a minor issue in case there is no .luacheckrc, I'll fix it and proceed myself. Thanks for the help!

**Pull request merged**